server/v2/stf/stf.go

```diff
@@ -56,10 +57,19 @@ func (s STF[T]) DeliverBlock(ctx context.Context, block appmanager.BlockRequest,
 		return nil, nil, err
 	}
 
+	events, valset, err := s.validatorUpdates(ctx, newState, block)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// append endblock events to the end of the block events
+	endBlockEvents = append(endBlockEvents, events...)
```

Comment:
I suppose my only gripe with this is that EndBlock events and ValSet update events are merged and treated as a single EndBlock events structure -- this is fine but just a bit a codesmell IMO. I guess the cleaner alternative is to have EndBlock return the valset events, but this is what we're trying to avoid, right?

Reply:
Im not a fan of this either, i can look into making it part of endblock. Outside of endblock it is another thing to implement. Do the endblock valset changes then allow stf to get the update at a later date. I think we can do it in one step. Ill look into this